server/src/main/java/server/Server.java

```diff
@@ -34,33 +34,35 @@
   private static MessageFinder messageFinder;
   private static MessagePoster messagePoster;
 
-  private static void setup() {
+  private static void setup() throws IOException{
     FileInputStream serviceAccount;
     FirebaseOptions firebaseOptions;
     StorageOptions storageOptions;
+    serviceAccount = new FileInputStream(Constants.FIREBASE_SERVICE_ACCOUNT_FILE);
 
     try {
 
-      serviceAccount = new FileInputStream(Constants.FIREBASE_SERVICE_ACCOUNT_FILE);
       firebaseOptions = FirebaseOptions
           .builder()
           .setCredentials(GoogleCredentials.fromStream(serviceAccount))
           .setDatabaseUrl(Constants.FIRESTORE_URL)
           .build();
-
       serviceAccount = new FileInputStream(Constants.FIREBASE_SERVICE_ACCOUNT_FILE);
       storageOptions = StorageOptions
           .newBuilder()
           .setProjectId(Constants.PROJECT_ID)
           .setCredentials(GoogleCredentials.fromStream(serviceAccount))
           .build();
+      serviceAccount.close();
```

Comment:
I don't _think_ we need this one, right? `finally` should happen no matter what

Reply:
fixed